lib/redis_client.rb

```diff
@@ -677,13 +678,14 @@ def ensure_connected(retryable: true)
           connection
         end
       rescue ConnectionError, ProtocolError => error
+        preferred_error = error if preferred_error.nil? || !error.is_a?(CircuitBreaker::OpenCircuitError)
```

Comment:
```suggestion
        preferred_error ||= error unless error.is_a?(CircuitBreaker::OpenCircuitError)
```

Reply:
Hum, actually, it's not exactly equivalent:

```suggestion
        preferred_error ||= error 
        preferred_error = error unless error.is_a?(CircuitBreaker::OpenCircuitError)
```